### PR TITLE
Fix the test path salmon/index and quant after switching to config data

### PR DESCRIPTION
**Pull request by BatoolMM**
Related to #363 to fix the test path for salmon/index and salmon/quant. However, when I run `nf-core modules create-test-yml`, I got an error                                                                     
                                                                                             
         `[-        ] process > test_salmon_index:SALMON_INDEX -                                                                                                                    
         executor >  local (1)                                                               
         [2c/9ffaab] process > test_salmon_index:SALMON_INDEX (transcri... [                 
         0%] 0 of 1   
          executor >  local (1)                                                               
         [2c/9ffaab] process > test_salmon_index:SALMON_INDEX (transcri... [                 
         0%] 0 of 1                                                                          
         Error executing process > 'test_salmon_index:SALMON_INDEX                           
         (transcriptome.fasta)'                                                              
                                                                                             
         Caused by:                                                                          
           Process `test_salmon_index:SALMON_INDEX (transcriptome.fasta)`                    
         terminated with an error exit status (134) `. 

It can be that I'm changing the path totally wrong 😕!

### CODE REVIEW

File: tests/software/salmon/index/main.nf [hunk 1]
```diff
@@ -5,8 +5,7 @@ nextflow.enable.dsl = 2
 include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addParams( options: [:] )
 
 workflow test_salmon_index {
-    genome        = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
-    transcriptome = file("${launchDir}/tests/data/genomics/sarscov2/genome/transcriptome.fasta", checkIfExists: true)
-
-    SALMON_INDEX ( genome, transcriptome )
+    def genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
```
Comment (drpatelh) [inline tests/software/salmon/index/main.nf:8]: We don't really need to use `def` here. I removed it for all the other modules but there is also no harm in having it.

File: tests/software/salmon/quant/main.nf [hunk 2]
```diff
@@ -6,26 +6,28 @@ include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addPar
...
-            ]
-    genome        = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
-    transcriptome = file("${launchDir}/tests/data/genomics/sarscov2/genome/transcriptome.fasta", checkIfExists: true)
-    gtf           = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.gtf", checkIfExists: true)
-
-    SALMON_INDEX ( genome, transcriptome )
-    SALMON_QUANT ( input, SALMON_INDEX.out.index, gtf, transcriptome, false )
+
+    def genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    def transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta', checkIfExists: true)
```
Comment (drpatelh) [inline tests/software/salmon/quant/main.nf:11]:
```suggestion
    def transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta]', checkIfExists: true)
```

Comment (drpatelh) [reply tests/software/salmon/quant/main.nf:11]: This is why the tests were failing 👀

File: tests/software/salmon/quant/main.nf [hunk 3]
```diff
@@ -6,26 +6,28 @@ include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addPar
...
+    def gtf   = file(params.test_data['sarscov2']['genome_gtf'], checkIfExists: true)
+    def input = [ [ id:'test', single_end:true ], // meta map
+                  file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
```
Comment (drpatelh) [inline tests/software/salmon/quant/main.nf:14]:
```suggestion
    input = [ [ id:'test', single_end:true ], // meta map
                file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) 
            ]
    genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta'], checkIfExists: true)
    gtf              = file(params.test_data['sarscov2']['genome_gtf'], checkIfExists: true)
    
```

File: tests/software/salmon/index/main.nf [hunk 4]
```diff
@@ -5,8 +5,7 @@ nextflow.enable.dsl = 2
...
+    def transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)
```
Comment (drpatelh) [inline tests/software/salmon/index/main.nf:9]:
```suggestion
    genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)
    
```

File: tests/software/salmon/quant/main.nf [hunk 5]
```diff
@@ -6,26 +6,28 @@ include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addPar
...
-
-    SALMON_INDEX ( genome, transcriptome )
-    SALMON_QUANT ( input, SALMON_INDEX.out.index, gtf, transcriptome, false )
+
+    def genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    def transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta', checkIfExists: true)
+    def gtf   = file(params.test_data['sarscov2']['genome_gtf'], checkIfExists: true)
+    def input = [ [ id:'test', single_end:false ], // meta map
+                  [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
+                    file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ] ]
```
Comment (drpatelh) [inline tests/software/salmon/quant/main.nf:28]:
```suggestion
    input = [ [ id:'test', single_end:false ], // meta map
              [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
            ]
    genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta'], checkIfExists: true)
    gtf              = file(params.test_data['sarscov2']['genome_gtf'], checkIfExists: true)
```

File: tests/software/salmon/quant/main.nf [hunk 6]
```diff
@@ -6,26 +6,31 @@ include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addPar
...
-                file("${launchDir}/tests/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz", checkIfExists: true) ] 
+              [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
+                file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
             ]
-    genome        = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
-    transcriptome = file("${launchDir}/tests/data/genomics/sarscov2/genome/transcriptome.fasta", checkIfExists: true)
-    gtf           = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.gtf", checkIfExists: true)
+    genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta'], checkIfExists: true)
+    gtf              = file(params.test_data['sarscov2']['genome_gtf'], checkIfExists: true)
```
Comment (drpatelh) [inline tests/software/salmon/quant/main.nf:31]:
```suggestion
    gtf              = file(params.test_data['sarscov2']['genome']['genome_gtf'], checkIfExists: true)
```

File: tests/software/salmon/quant/main.nf [hunk 7]
```diff
@@ -6,26 +6,31 @@ include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addPar
...
-              file("${launchDir}/tests/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true) 
+                file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) 
             ]
-    genome        = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
-    transcriptome = file("${launchDir}/tests/data/genomics/sarscov2/genome/transcriptome.fasta", checkIfExists: true)
-    gtf           = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.gtf", checkIfExists: true)
+    genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta'], checkIfExists: true)
+    gtf              = file(params.test_data['sarscov2']['genome_gtf'], checkIfExists: true)
+
```
Comment (drpatelh) [inline tests/software/salmon/quant/main.nf:16]:
```suggestion
    gtf              = file(params.test_data['sarscov2']['genome']['genome_gtf'], checkIfExists: true)
```

File: tests/software/salmon/quant/main.nf [hunk 8]
```diff
@@ -6,26 +6,30 @@ include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addPar
...
     input = [ [ id:'test', single_end:true ], // meta map
-              file("${launchDir}/tests/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true) 
+                file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) 
             ]
-    genome        = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
-    transcriptome = file("${launchDir}/tests/data/genomics/sarscov2/genome/transcriptome.fasta", checkIfExists: true)
-    gtf           = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.gtf", checkIfExists: true)
+    genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta'], checkIfExists: true)
```
Comment (drpatelh) [inline tests/software/salmon/quant/main.nf:14]:
```suggestion
    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)
```

File: tests/software/salmon/quant/main.nf [hunk 9]
```diff
@@ -6,26 +6,30 @@ include { SALMON_INDEX } from '../../../../software/salmon/index/main.nf' addPar
...
-              [ file("${launchDir}/tests/data/genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
-                file("${launchDir}/tests/data/genomics/sarscov2/illumina/fastq/test_2.fastq.gz", checkIfExists: true) ] 
+              [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
+                file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true) ]
             ]
-    genome        = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.fasta", checkIfExists: true)
-    transcriptome = file("${launchDir}/tests/data/genomics/sarscov2/genome/transcriptome.fasta", checkIfExists: true)
-    gtf           = file("${launchDir}/tests/data/genomics/sarscov2/genome/genome.gtf", checkIfExists: true)
+    genome_fasta     = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
+    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome.fasta'], checkIfExists: true)
```
Comment (drpatelh) [inline tests/software/salmon/quant/main.nf:29]:
```suggestion
    transcript_fasta = file(params.test_data['sarscov2']['genome']['transcriptome_fasta'], checkIfExists: true)
```

**Comment by drpatelh** [comment]
Ok. The modules are now running @BatoolMM 😅 A few typos and missing braces. You just need to `git pull` the changes I made first and then you just need to do a find and replace for `/salmon/salmon/` -> `/index/salmon/` in [this file](https://github.com/nf-core/modules/blob/master/tests/software/salmon/quant/test.yml) and it should get the tests passing. I would have changed it but that file isn't in the current git diff.

File: tests/software/salmon/index/main.nf (hunk 1 above)
Comment (BatoolMM) [reply tests/software/salmon/index/main.nf]: I see, I'm getting the hang of it now. Thank you!

File: tests/software/salmon/quant/main.nf (hunk 2 above)
Comment (BatoolMM) [reply tests/software/salmon/quant/main.nf]: Good catch  😲 !

**Comment by BatoolMM** [comment]
Waw, SO MANY typos 😱, you have so much patience to fix them all, very kind of you and sorry for that.  
Can I ask why would you change `/salmon/salmon/` to `/index/salmon/`. It tried to generate this `test.yml` with `nf-core modules create-test-yml` but still giving me the same error. I suspect it's something to do with my local machine, so, I'll try to figure out this error later or ask in Slack if the problem persisted. 

I already applied the changes by replacing `/salmon/salmon/` -> `/index/salmon/`. This PR is now passing all the CI, thanks again 🌸 !

**Comment by drpatelh** [comment]
No worries! It was due to recent changes in the default output folder for the module:
https://github.com/nf-core/modules/pull/336

Thanks!

**Review by drpatelh — APPROVED** [review]

**Pull request merged**